interfaces/builtin/adb_support.go

```diff
@@ -0,0 +1,205 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const adbSupportSummary = `allows access to android devices for fastboot and adb`
+
+const adbSupportBaseDeclarationSlots = `
+  adb-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+var adbSupportConnectedPlugUDev = []string{
```

Comment:
Please add a comment like so:
```
// These udev rules are specifically for making it convenient for people
// to connect to their device without requiring root privileges. Upstream
// adb also uses 'GROUP="plugdev"', but this group is not guaranteed
// to exist and since MODE is 0666, it isn't required. We use 'MODE="0666"'
// since the access to the USB devices doesn't provide a privilege
// escalation on the host and the connected Android device has its own
// checks for allowing access that are independent of the host USB
// setup, therefore a more restricted setup (eg 'MODE="0660"') doesn't
// appreciably improve security.
```

Reply:
Oddly I cannot respond to your comment about raw-usb above so I will ask my question here. How would this interface work without raw-usb? (apart from adb-as-a-service which itself would use raw-usb).

Reply:
Someone needs to profile (ie, apparmor) how adb operates without plugging raw-usb. One way to do this would be for someone (eg, @NeoTheThird) to install a snap in devmode with raw-usb disconnected and then pasting the ALLOWED policy violations from journald.